src/OrchardCore.Modules/OrchardCore.Cors/Services/CorsOptionsConfiguration.cs

```diff
@@ -0,0 +1,84 @@
+using Microsoft.AspNetCore.Cors.Infrastructure;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System.Linq;
+
+namespace OrchardCore.Cors.Services
+{
+    public class CorsOptionsConfiguration : IConfigureOptions<CorsOptions>
+    {
+        private readonly CorsService _corsService;
+        private readonly ILogger<CorsOptionsConfiguration> _logger;
+
+        public CorsOptionsConfiguration(CorsService corsService, ILogger<CorsOptionsConfiguration> logger)
+        {
+            _corsService = corsService;
+            _logger = logger;
+        }
+
+        public void Configure(CorsOptions options)
+        {
+            var corsSettings = _corsService.GetSettingsAsync().GetAwaiter().GetResult();
+            if (corsSettings?.Policies == null || !corsSettings.Policies.Any())
+                return;
+
+            foreach (var corsPolicy in corsSettings.Policies)
+            {
+                if (corsPolicy.AllowCredentials && corsPolicy.AllowAnyOrigin)
+                {
+                    _logger.LogWarning($"AllowCredentials and AllowAnyOrigin is considered a security risk, policy {corsPolicy.Name} not loaded");
```

Comment:
Bad logging usage

Reply:
@MatthijsKrempel 

@sebastienros means that in place of using string interpolation, the logger allows to pass parameter and parse them efficiently when it is called.

    _logger.LogWarning("AllowCredentials ... policy {PolicyName} not loaded", corsPolicy.Name);